### PR TITLE
Pytest conversion for tests/foreman/cli/test_report.py

### PR DESCRIPTION
**Pull request by jameerpathan111**
Test result:
```
$ pytest tests/foreman/cli/test_report.py 
============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jpathan/projects/robottelo, configfile: pyproject.toml
plugins: ibutsu-1.13, services-2.2.1, mock-3.5.1, reportportal-5.0.7, xdist-2.2.1, forked-1.3.0
collected 3 items

tests/foreman/cli/test_report.py ...                                     [100%]

========================= 3 passed in 69.63s (0:01:09) =========================
 ```

### CODE REVIEW

File: tests/foreman/cli/test_report.py [hunk 1]
```diff
@@ -23,67 +23,63 @@
 from robottelo import ssh
 from robottelo.cli.base import CLIReturnCodeError
 from robottelo.cli.report import Report
-from robottelo.test import CLITestCase
 
 
-class ReportTestCase(CLITestCase):
-    """Test class for Reports CLI. """
+@pytest.fixture(scope='module')
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_report.py:28]: You could make this an auto use fixture and then remove it from each tests argspec.

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_report.py (hunk 1 above)
Comment (jameerpathan111) [reply tests/foreman/cli/test_report.py]: updated the pr.

File: tests/foreman/cli/test_report.py [hunk 2]
```diff
@@ -23,67 +23,63 @@
...
-        super().setUp()
-        self.run_puppet_agent()
 
-    def run_puppet_agent(self):
-        """Retrieves the client configuration from the puppet master and
-        applies it to the local host. This is required to make sure
-        that we have reports available.
-        """
-        ssh.command('puppet agent -t')
+@pytest.mark.tier1
```
Comment (pondrejk) [inline tests/foreman/cli/test_report.py:37]: not sure if this test is needed, we don't assert on anything here + Report.list() is run in both remaining tests, maybe a chance for removal

Comment (jameerpathan111) [reply tests/foreman/cli/test_report.py:37]: makes sense. removed the test.

**Review by JacobCallahan — APPROVED** [review]
ACK, looks great!

File: tests/foreman/cli/test_report.py (hunk 2 above)
Comment (omkarkhatavkar) [reply tests/foreman/cli/test_report.py:37]: this has been resolved

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Pull request merged**